src/JuMP.jl

```diff
@@ -517,18 +518,18 @@ Base.copy(c::SDPConstraint, new_model::Model) =
 ##########################################################################
 # ConstraintRef
 # Reference to a constraint for retrieving solution info
-immutable ConstraintRef{T<:JuMPConstraint}
-    m::Model
+immutable ConstraintRef{M<:AbstractModel,T<:JuMPConstraint}
```

Comment:
An alternative approach would be to make this ``AbstractConstraintRef`` and make
```jl
typealias ConstraintRef{T} AbstractConstraintRef{Model,T}
```
This would require less code changes in the rest of the PR.